src/shared/protocol.ts

```diff
@@ -87,6 +87,17 @@ export type RequestOptions = {
    * May be used to indicate to the transport which incoming request to associate this outgoing request with.
    */
   relatedRequestId?: RequestId;
+  /**
+   * The last event ID to associate with the request.
+   * Used to resume long-running requests that were interrupted when creating a new client instance.
+   */
+  lastEventId?: string;
+
+  /**
+   * A callback that is invoked when the last event ID is updated.
+   * This is used to notify the client that the last event ID has changed, so the client can update its state accordingly.
+   */
+  onLastEventIdUpdate?: (event: string) => void;
```

Comment:
IMO we should generalize this to terms and concepts that aren't limited to the Streamable HTTP transport.

For example, we could refer to a general concept of resumability:

```suggestion

  /**
   * The resumption token used to continue long-running requests that were interrupted.
   *
   * This allows clients to reconnect and continue from where they left off, if supported by the transport.
   */
  resumptionToken?: string;
  
  /**
   * A callback that is invoked when the resumption token changes, if supported by the transport.
   *
   * This allows clients to persist the latest token for potential reconnection.
   */
  onresumptiontoken?: (token: string) => void;
```

Yes, it's _currently_ only implemented for Streamable HTTP, but this at least leaves the API surface forward-compatible with other transports that might be resumable in future.